app/src/frontend/building/data-container.tsx

```diff
@@ -14,39 +41,55 @@ import InfoBox from '../components/info-box';
  *
  * @param WrappedComponent
  */
-const withCopyEdit = (WrappedComponent) => {
-    return class extends React.Component<any, any> { // TODO: add proper types
+const withCopyEdit = (WrappedComponent: React.ComponentType<CategoryViewProps>) => {
```

Comment:
Through the interface `CategoryViewProps` and the `React.ComponentType` generic type, we provide type checking for the uses of `withCopyEdit`, and better type checking for uses of `WrappedComponent` inside the body of the HOC